src/main/java/org/openrewrite/java/testing/search/FindUnitTests.java

```diff
@@ -0,0 +1,118 @@
+            @Override
+            public J visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+                // get the method declaration the method invocation is in
+                J.MethodDeclaration methodDeclaration = getCursor().firstEnclosing(J.MethodDeclaration.class);
+                if (methodDeclaration != null
+                        && methodDeclaration.getLeadingAnnotations().stream()
+                        .filter(o -> o.getAnnotationType() instanceof J.Identifier)
+                        .anyMatch(o -> "Test".equals(o.getSimpleName()))) {
+                    UnitTest unitTest = new UnitTest(
+                            getCursor().firstEnclosingOrThrow(J.ClassDeclaration.class).getType().getFullyQualifiedName(),
+                            methodDeclaration.getSimpleName(),
+                            methodDeclaration.printTrimmed(getCursor()));
+                    acc.unitTestAndTheirMethods.merge(unitTest,
+                            new HashSet<>(singletonList(method)),
+                            (a, b) -> {
+                                a.addAll(b);
+                                return a;
+                            });
+                }
+                return super.visitMethodInvocation(method, ctx);
+            }
+        };
+        return Preconditions.check(new IsLikelyTest().getVisitor(), scanningVisitor);
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor(Accumulator acc) {
+        JavaVisitor<ExecutionContext> tableRowVisitor = new JavaVisitor<ExecutionContext>() {
+            @Override
+            public J visitMethodDeclaration(J.MethodDeclaration methodDeclaration, ExecutionContext ctx) {
+                for (UnitTest unitTest : acc.unitTestAndTheirMethods.keySet()) {
+                    for (J.MethodInvocation method : acc.unitTestAndTheirMethods.get(unitTest)) {
+                        if (method.getSimpleName().equals(methodDeclaration.getSimpleName())) {
+                            unitTestTable.insertRow(ctx, new FindUnitTestTable.Row(
+                                    methodDeclaration.getName().toString(),
+                                    methodDeclaration.getSimpleName(),
+                                    method.printTrimmed(getCursor()),
+                                    unitTest.getClazz(),
+                                    unitTest.getUnitTestName()
+                            ));
+                        }
+                    }
+                    return super.visitMethodDeclaration(methodDeclaration, ctx);
+                }
+                return super.visitMethodDeclaration(methodDeclaration, ctx);
+            }
+        };
+        return Preconditions.check(new IsLikelyNotTest().getVisitor(), tableRowVisitor);
+    }
+
+}
+
+@Value
+class UnitTest {
+    String clazz;
+    String unitTestName;
+    String unitTest;
```

Comment:
Instead of String we might want to use the types of the LST elements matched here instead.